pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -375,6 +377,12 @@ public void addFailed(ManagedLedgerException exception, Object ctx) {
         return completableFuture;
     }
 
+    private void checkAppendMarkerException(ManagedLedgerException exception) {
+        if (exception instanceof ManagedLedgerException.ManagedLedgerAlreadyClosedException) {
+            topic.getManagedLedger().readyToCreateNewLedger();
```

Comment:
Do we need to check the state is `State.WriteFailed`? 

```
else if (state == State.Closed) {
            addOperation.failed(new ManagedLedgerAlreadyClosedException("Managed ledger was already closed"));
            return;
        } else if (state == State.WriteFailed) {
            addOperation.failed(new ManagedLedgerAlreadyClosedException("Waiting to recover from failure"));
            return;
        }
```

We also remove in this PR https://github.com/apache/pulsar/pull/14738/files#diff-4fb66809c520962f28616ddcab91aca0abb7aa04ee3ef0703097cf5976080ba8L166-L168

@congbobo184 Do you remember the context?

Reply:
`topic.getManagedLedger().readyToCreateNewLedger();` this method is 
```
if (STATE_UPDATER.compareAndSet(this, State.WriteFailed, State.ClosedLedger)){
           log.info("[{}] Managed ledger is now ready to accept writes again", name);
       }
```

so we don't need to check the state is `State.WriteFailed`

Reply:
Sorry, I mean if we get a `ManagedLedgerAlreadyClosedException`, it can be `State.Closed` and `State.WriteFailed`. We only need to call `readyToCreateNewLedger()` for `State.WriteFailed` but not the `State.Closed`? Because `State.Closed` means the managed ledger is closed, it not equals to the `State.ClosedLedger`

Reply:
yeah, we only need to readyToCreateNewLedger() in `State.WriteFiled`. If it  in `State.Closed`, we can't update `State.WriteFiled` to `State.ClosedLedger`. Because method `readyToCreateNewLedger()` has already check the `State.WriteFiled`, so we don't need to check twice